packages/evm/src/opcodes/gas.ts

```diff
@@ -166,7 +166,8 @@ export const dynamicGasHandlers: Map<number, AsyncDynamicGasHandler | SyncDynami
         let charge2929Gas = true
         if (
           common.isActivatedEIP(6800) &&
-          runState.interpreter._evm.getPrecompile(address) === undefined
+          runState.interpreter._evm.getPrecompile(address) === undefined &&
+          !address.equals(createAddressFromStackBigInt(common.param('systemAddress')))
```

Comment:
My only concern with this is that should consider the larger picture, mentioned by @g11tech, that it makes sense to store system contract address, codehash, and code, in memory the way we do for precompiles.  I think we should go ahead and merge this as is but then do some work to make the system contract (plus any future ones) available in memory so we don't need to read code from storage for them.

Reply:
yea we need to address this both for system calls which will have address equal to system address as well as normal contract calls into system contracts